tests.md

```diff
@@ -0,0 +1,833 @@
+    describe "Wallet names valid"
+    describe "Wallet names invalid"
+    describe "Pub Key invalid"
+    describe "Address pool gap invalid"
+
+spec = describe "BYRON_HW_WALLETS"
+    it "HW_WALLETS_01 - Restoration from account public key preserves funds"
+  describe "HW_WALLETS_03 - Cannot do operations requiring private key"
+        it "Cannot send tx"
+        it "Cannot update pass"
+  describe "HW_WALLETS_04 - Can manage HW wallet the same way as others"
+        it "Can update name"
+        it "Can get tx fee"
+        it "Can delete"
+        it "Can see utxo"
+        it "Can list addresses"
+        it "Can have address pool gap"
+        it "Can list transactions"
+        it "Can get coin selection"
+  describe "HW_WALLETS_05 - Wallet from pubKey is available"
+        it "Can get wallet"
+        it "Can list wallet"
+        it "The same account and mnemonic wallet can live side-by-side"
+
+spec = describe "NEW_SHELLEY_TRANSACTIONS"
+    it "TRANS_NEW_JOIN_01b - Invalid pool id"
+    it "TRANS_NEW_JOIN_01b - Absent pool id"
+    it "TRANS_NEW_JOIN_01c - Multidelegation not supported"
+    it "TRANS_NEW_JOIN_01d - Multiaccount not supported"
+    it "TRANS_NEW_JOIN_01e - Can re-join and withdraw at once"
+    it "TRANS_NEW_JOIN_01f - Cannot re-join the same pool in Babbage"
+    it "TRANS_NEW_JOIN_01f - Can re-join the same pool in Conway"
+    it "TRANS_NEW_QUIT_01 - Cannot quit if not joined"
+```
+
+## E2E Tests
+
+the following tests are actually not about cardano-wallet at all but testing other utilities: [cardano-address]()
+They also depend on code defined in [cardano-wallet-rb](https://github.com/piotr-iohk/cardano-wallet-rb)
+
+**They can be safely discarded**
+
+```
+misc:
+     describe CardanoWallet::Misc::Node do
+       it 'Can get latest block header info' do
+     describe CardanoWallet::Misc::Network do
+       it 'Can get network information' do
+       it 'Can check network clock offset' do
+       it 'Can check network parameters' do
+         it 'SMASH health - unreachable' do
+         it 'SMASH health - bad url' do
+       it 'Inspect invalid address' do
+       it 'Inspect Shelley payment address' do
+       it 'Inspect Shelley stake address' do
+       it 'Inspect Byron Random address' do
+       it 'Inspect Byron Icarus address' do
+         it 'Enterprise script address - signature' do
+         it 'Enterprise script address - any' do
+         it 'Enterprise script address - all' do
+         it 'Enterprise script address - some' do
+         it 'Reward account script address - any' do
+         it 'Delegating script address - any' do
+         it 'Enterprise pub key address' do
+         it 'Reward account pub key address' do
+         it 'Delegating address with both pub key credentials' do
+         it 'Delegating address - payment from script, stake from key' do
+         it 'Delegating address - payment from key, stake from script' d
+       it 'Malformed payload when tx is not binary' do
+```
```

Comment:
The test(s)

```ruby
it "I can read and update settings to #{strategy} and verify SMASH health = #{smash_health_response}"
```

seems to be missing from this list, even though they are part of the file `misc_spec.rb`.

I'm don't know whether we have an equivalent of this test in Haskell. Testing the interaction with the SMASH metadata server is relevant for Daedalus, as it affects the display of token information.